File: LeetCode/Contains Duplicate/ContainsDuplicate.java
class ContainsDuplicate {
//     public boolean containsDuplicate(int[] nums) {
//         Arrays.sort(nums);
        
//         for(int i=0; i < nums.length-1; i++){
//             if(nums[i] == nums[i+1])
//                 return true;
//         }
//         return false;
//     }
    
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> numbers = new HashSet<>();
        
        for(int i=0; i<nums.length; i++){
            if(numbers.contains(nums[i])){
                return true;            
            }
            numbers.add(nums[i]);
        }
        return false;
    }
    
}

File: NeetCode/ArraysAndHash/ContainsDuplicate.py
"""
LeetCode: 217. Contains Duplicate
NeetCode: Blind 1/150

Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.
"""
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        dup_dict = set()
        
        for n in nums:
            if(n in dup_dict):
                return True
            dup_dict.add(n)
        return False
